tests/test_portchannel.py

```diff
@@ -57,3 +57,130 @@ def test_Portchannel(self, dvs, testlog):
+        tbl.set("PortChannel002|40.0.0.2/31", fvs)
+        time.sleep(1)
+        tbl.set("PortChannel003", fvs)
+        tbl.set("PortChannel003|40.0.0.4/31", fvs)
+        time.sleep(1)
+        tbl.set("PortChannel004", fvs)
+        tbl.set("PortChannel004|40.0.0.6/31", fvs)
+        time.sleep(1)
+
+        # check application database
+        tbl = swsscommon.Table(self.pdb, "INTF_TABLE:PortChannel001")
+        intf_entries = tbl.getKeys()
+        assert len(intf_entries) == 1
+        assert intf_entries[0] == "40.0.0.0/31"
+        tbl = swsscommon.Table(self.pdb, "INTF_TABLE:PortChannel002")
+        intf_entries = tbl.getKeys()
+        assert len(intf_entries) == 1
+        assert intf_entries[0] == "40.0.0.2/31"
+        tbl = swsscommon.Table(self.pdb, "INTF_TABLE:PortChannel003")
+        intf_entries = tbl.getKeys()
+        assert len(intf_entries) == 1
+        assert intf_entries[0] == "40.0.0.4/31"
+        tbl = swsscommon.Table(self.pdb, "INTF_TABLE:PortChannel004")
+        intf_entries = tbl.getKeys()
+        assert len(intf_entries) == 1
+        assert intf_entries[0] == "40.0.0.6/31"
+
+        dvs.runcmd("arp -s 40.0.0.1 00:00:00:00:00:01")
+        time.sleep(1)
+        dvs.runcmd("arp -s 40.0.0.3 00:00:00:00:00:03")
+        time.sleep(1)
+        dvs.runcmd("arp -s 40.0.0.5 00:00:00:00:00:05")
+        time.sleep(1)
+        dvs.runcmd("arp -s 40.0.0.7 00:00:00:00:00:07")
+        time.sleep(1)
+
+        ps = swsscommon.ProducerStateTable(self.pdb, "ROUTE_TABLE")
+        fvs = swsscommon.FieldValuePairs([("nexthop","40.0.0.1,40.0.0.3,40.0.0.5,40.0.0.7"), ("ifname", "PortChannel001,PortChannel002,PortChannel003,PortChannel004")])
+
+        ps.set("2.2.2.0/24", fvs)
+        time.sleep(1)
+
+        # check if route has propagated to ASIC DB
+        re_tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY")
+
+        found_route = False
+        for key in re_tbl.getKeys():
+            route = json.loads(key)
+            print route
+            if route["dest"] == "2.2.2.0/24":
+               found_route = True
+               break
+
+        assert found_route
+
+        # check if route points to next hop group
+        nhg_tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP")
+        (status, fvs) = re_tbl.get(key)
+        for v in fvs:
+            if v[0] == "SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID":
+                nhg_id = v[1]
+
+        (status, fvs) = nhg_tbl.get(nhg_id)
+        assert status
+
+        # check if next hop group consists of 4 members
+        nhg_member_tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP_MEMBER")
+        keys = nhg_member_tbl.getKeys()
+        assert len(keys) == 4
+
+        for key in keys:
+            (status, fvs) = nhg_member_tbl.get(key)
+            for v in fvs:
+                if v[0] == "SAI_NEXT_HOP_GROUP_MEMBER_ATTR_NEXT_HOP_GROUP_ID":
+                    assert v[1] == nhg_id
```

Comment:
I assume it is a partial commit and you are modifying the test to simulate oper down scenario.